### PR TITLE
domain: support retry if load failed.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @qiuyesuifeng

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -142,43 +142,57 @@ func (do *Domain) tryReload() {
...
+		}
+
+		done <- err
+	}()
+
+	select {
+	case err := <-done:
+		return errors.Trace(err)
+	case <-time.After(timeout):
+		return errors.Errorf("reload schema timeout")
```
Comment (qiuyesuifeng) [inline domain/domain.go]: errors.New

File: domain/domain.go [hunk 2]
```diff
@@ -142,43 +142,57 @@ func (do *Domain) tryReload() {
...
+		for {
+			err = kv.RunInNewTxn(do.store, false, do.loadInfoSchema)
+			// if err is db closed, we will return it directly, otherwise, we will
+			// check reloading again.
+			if terror.ErrorEqual(err, localstore.ErrDBClosed) {
+				break
+			}
+
+			if err != nil {
+				// TODO: use a backoff algorithm.
```
Comment (qiuyesuifeng) [inline domain/domain.go]: Add a trace log?

Comment (ngaut) [inline domain/domain.go]: log a warning message.

Comment (siddontang) [inline domain/domain.go]: I think using a error level message is better, because this error seems serious.

File: domain/domain.go [hunk 3]
```diff
@@ -142,43 +142,58 @@ func (do *Domain) tryReload() {
...
+		for {
+			err = kv.RunInNewTxn(do.store, false, do.loadInfoSchema)
+			// if err is db closed, we will return it directly, otherwise, we will
+			// check reloading again.
+			if terror.ErrorEqual(err, localstore.ErrDBClosed) {
+				break
+			}
+
+			if err != nil {
+				log.Errorf("load schema err %v, retry again", err)
```
Comment (ngaut) [inline domain/domain.go]: Does tests cover this condition?

Comment (siddontang) [inline domain/domain.go]: No, it is very hard to construct the error case.

**Comment by ngaut** [comment]
LGTM

**Pull request merged**